merge_sort/merge_sort.py

```diff
@@ -0,0 +1,68 @@
+def merge(array, left, right):
+    """
+    Perform Merge Operation between arrays.
+    Time Complexity: Theta(nLogn)
+    Auxiliary Space: O(n)
+    :param array: Iterable of elements
+    :param left: left limit for merge sort
+    :param right: right limit for merge sort
+    :return: no returns, merges arrays.
+    """
+    mid = int((left + right) / 2)
+    s1 = int(mid - left + 1)
+    s2 = int(right - mid)
+    l = [0] * s1
+    r = [0] * s2
+    for i in range(s1):
+        l[i] = array[int(left + i)]
+
+    for j in range(s2):
+        r[j] = array[mid + 1 + j]
+
+    i = j = int(0)
+    k = int(left)
+    while(i < s1 and j < s2):
+        if(l[i] <= r[j]):
+            array[k] = l[i]
+            i = i + 1
+        else:
+            array[k] = r[j]
+            j = j + 1
```

Comment:
You could simply reset `array` and append things as they come. This and the following stuff can become better if you use `pop()`s for removing elements from `l` and `r` and keep adding them to `array` while both of them are not empty.

You may need to return `array` based on how you are performing the refilling.